qiita_db/study.py

```diff
@@ -347,6 +348,56 @@ def create(cls, owner, title, efo, info, investigation=None):
 
         return cls(study_id)
 
+    @classmethod
+    def delete(cls, id_):
+        r"""Deletes the study from the database
+
+        Parameters
+        ----------
+        id_ : integer
+            The object identifier
+
+        Raises
+        ------
+        QiitaDBError
+            If the sample_(id_) table exists means a sample template exists
+        """
+        cls._check_subclass()
+
+        conn_handler = SQLConnectionHandler()
+        if exists_table('sample_%d' % id_, conn_handler):
+            raise QiitaDBError('Study "%s" cannot be erased because it has a '
+                               'sample template' % cls(id_).title)
+
+        queue = "delete_study_%d" % id_
+        conn_handler.create_queue(queue)
+
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.study_sample_columns WHERE study_id = %s",
+            (id_, ))
+
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.study_experimental_factor WHERE study_id = %s",
+            (id_, ))
+
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.study_pmid WHERE study_id = %s", (id_, ))
+
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.study_environmental_package WHERE study_id = "
+            "%s", (id_, ))
+
+        conn_handler.add_to_queue(
+            queue,
+            "DELETE FROM qiita.study WHERE study_id = %s", (id_, ))
+
+        conn_handler.execute_queue(queue)
```

Comment:
I'm surprised that this is actually working. It should fail since you are not removing the row from the `study_users` and `investigation_study` table. Does this mean that we are missing a FK on these tables?